js/exporter/exceljs/export_pivot_grid.js

```diff
@@ -1,47 +1,238 @@
+        this.columnFieldHeaders = this._tryGetFieldHeaders('column');
+        this.rowFieldHeaders = this._tryGetFieldHeaders('row');
+    }
+
+    _getFirstColumnIndex() {
+        return this.topLeftCell.column;
+    }
+
+    _getWorksheetFrozenState(cellRange) {
+        const { x, y } = this.dataProvider.getFrozenArea();
+
+        return {
+            state: 'frozen',
+            xSplit: cellRange.from.column + x - 1,
+            ySplit: cellRange.from.row + y + this._getFieldHeaderRowsCount() - 1,
+        };
+    }
+
+    _getFieldHeaderRowsCount() {
+        let result = 0;
+
+        if(this._allowExportFilterFieldHeaders()) {
+            result++;
+        }
+
+        if(this._allowExportDataFieldHeaders() || this._allowExportColumnFieldHeaders()) {
+            result++;
+        }
+
+        return result;
+    }
 
     _getCustomizeCellOptions(excelCell, pivotCell) {
         return {
             excelCell: excelCell,
             pivotCell: pivotCell
         };
-    },
+    }
 
     _isFrozenZone() {
         return true;
-    },
+    }
+
+    _isHeaderCell(rowIndex, cellIndex) {
+        return rowIndex < this.dataProvider.getColumnAreaRowCount() || cellIndex < this.dataProvider.getRowAreaColCount();
+    }
+
+    _getDefaultFieldHeaderCellsData(value) {
+        return {
+            text: value,
+            value: value,
+        };
+    }
+
+    _isInfoCell(rowIndex, cellIndex) {
+        return rowIndex < this.dataProvider.getColumnAreaRowCount() && cellIndex < this.dataProvider.getRowAreaColCount();
+    }
+
+    _allowToMergeRange(rowIndex, cellIndex, rowspan, colspan) {
+        return !((this.dataProvider.isColumnAreaCell(rowIndex, cellIndex) && !this.mergeColumnFieldValues && !!colspan)
+        || (this.dataProvider.isRowAreaCell(rowIndex, cellIndex) && !this.mergeRowFieldValues && !!rowspan));
+    }
+
+    _trySetAutoFilter() {}
+
+    _trySetFont(excelCell, bold) {
+        if(isDefined(bold)) {
+            excelCell.font = excelCell.font || {};
+            excelCell.font.bold = bold;
+        }
+    }
+
+    _getFieldHeaderStyles() {
+        // eslint-disable-next-line spellcheck/spell-checker
+        const borderStyle = { style: 'thin', color: { argb: 'FF7E7E7E' } };
+
+        return {
+            alignment: getDefaultAlignment(this.dataProvider._options.rtlEnabled),
+            bold: true,
+            border: {
+                bottom: borderStyle,
+                left: borderStyle,
+                right: borderStyle,
+                top: borderStyle
+            }
+        };
+    }
+
+    _trySetOutlineLevel() {}
+
+    _getAllFieldHeaders() {
+        return this.dataProvider._exportController.getDataSource()._descriptions;
+    }
+
+    _tryGetFieldHeaders(area) {
+        if(!this[`export${camelize(area, true)}FieldHeaders`]) {
+            return [];
+        }
+
+        let fields = this._getAllFieldHeaders()[area === 'data' ? 'values' : `${area}s`].filter(r => r.area === area);
```

Comment:
what is `r` here?
```javascript
let fields = this._getAllFieldHeaders()[area === 'data' ? 'values' : `${area}s`]
    .filter(fieldHeader => fieldHeader.area === area);
```

Reply:
`r` is header data view. I am going concrete this in next commit